esp/esdllib/README.md

```diff
@@ -701,9 +787,252 @@ Is not applied to service methods with `@querytype` values:
+```
+
+Similarly, we can use a predicate to select the second occurrence of the `mi` node:
+
+```xml
+  <es:set-value target="result" select="foo/bar/mi[$twoVar]"/>
+```
+
+Yielding:
+
+```xml
+  <result>call myself</result>
+```
+
+Or we could use a predicate to select the `do` node with a `name` attribute equal to *deer*:
 
-Additionally, we support a subset of EXSLT-Functions (Extended XSLT) for use in XPath expressions. The exslt functions are grouped into modules of related features like *strings*, *sets* and *math*. There are different implmentations of the modules and not all implementations support all functions. We are using the `libxslt` implementation. The reference at [exslt.org](http://exslt.org/func/index.html) lists the functions in each module and which implmenetations support each function. See below for details of our supported modules
+```xml
+  <es:set-value target="result" select="foo/bar/do[@name=$deerVar]"/>
+```
+
+Yielding:
+
+```xml
+  <result>a female deer</result>
+```
+
+#### Expressions and Function Arguments
+
+Variables can also be referenced in *Expressions* and in function *Arguments*. At its simplest an *Expression* can be a variable reference. This example shows setting the `result` node to the value of a `deerVar`:
+
+```xml
+  <es:set-value target="result" select="$deerVar"/>
+```
+
+Yields this:
+
+```xml
+  <result>deer</result>
+```
+
+This more complex expression with a variable:
+
+```xml
+  <es:set-value target="result" select="$twoVar + 10"/>
+```
+
+Yields:
+
+```xml
+  <result>12</result>
+```
+
+Variables can also be referenced in *Arguments* to functions:
+
+```xml
+  <es:set-value target="result" select="string-length($deerVar) + 10"/>
+```
+
+Yielding:
+
+```xml
+  <result>14</result>
+```
+
+### Extensions
+
+ESDL Scripts extend standard XPath features to make setting entry point default `@source` and `@target` nodes easier. In the root of any entry point, the `@source` and `@target` XPaths may contain the [Implicit Variables](#implicit-variables) as *Axes*. These special variable references are wrapped in curly braces.
+
+For example, say we have the following simplified request bound for a roxie backend that we're going to manipulate with a `BackendRequest` entry point script:
+
+```xml
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <MyMethodModule.MyMethodQuery>
+      <Context>...</Context>
+      <MyMethodRequest>
+        <Row>
+          <Options></Options>
+        </Row>
+      </MyMethodRequest>
+    </MyMethodModule.MyMethodQuery>
+  </soap:Body>
+</soap:Envelope>
+```
+
+The goal is to have a single script for a service that adds several nodes to the `Options` section for every method in the service. Each method has a different request roxie query name (here `MyMethodModule.MyMethodQuery`) and different method request name (here `MyMethodRequest`). This script will correctly set the default target for all methods:
```

Comment:
"different request roxie query" seems... off. I think "request" can be dropped, but could also see it being changed to "requested".

Reply:
fixed. - I think just an editing typo from previous revision.